### PR TITLE
Update Alpine to v3.16

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- Update base image to Alpine v3.16
- Update runc to v1.1.2 (fix CVE-2022-29162)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -49,23 +49,17 @@ require (
...
-// Fix GO-2021-0085
-// Fix GO-2021-0087
-replace github.com/opencontainers/runc => github.com/opencontainers/runc v1.0.3
+// Fix CVE-2022-29162
+replace github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.2
 
-// Fix CVE-2021-41190
-replace github.com/opencontainers/image-spec => github.com/opencontainers/image-spec v1.0.2
+// Fix CVE-2022-27191
+replace golang.org/x/crypto => golang.org/x/crypto v0.0.0-20220411220226-7b82a4e95df4
```
Comment (hiddeco) [inline go.mod:62]: Latest available at this moment is:

```go
replace golang.org/x/crypto => golang.org/x/crypto v0.0.0-20220518034528-6f7dac969898
```

**Review by makkes — APPROVED** [review]

File: api/go.mod [hunk 2]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/fluxcd/pkg/apis/meta v0.14.0
 	k8s.io/apiextensions-apiserver v0.24.0
 	k8s.io/apimachinery v0.24.0
-	sigs.k8s.io/controller-runtime v0.11.2
+	sigs.k8s.io/controller-runtime v0.12.1
```
Comment (stefanprodan) [inline api/go.mod:10]: Please undo this change, controller-runtime v0.12.1 breaks the logging and has serious bugs!

Comment (hiddeco) [reply api/go.mod:10]: Can you elaborate on the bugs? I personally find "breaks logging" not a very strong arguments to stall further updates. As the further set of changes seems to be good: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.12.0

Comment (hiddeco) [reply api/go.mod:10]: Discussed this further. We need to look into our glue around logging before progressing this update.

Comment (hiddeco) [reply api/go.mod:10]: For further reference, we need https://github.com/kubernetes-sigs/controller-runtime/pull/1911 to be included in a `0.12.x` release.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**